### ISSUE TITLE
Make user data and configuration directory/ies XDG-compliant.

### ISSUE THREAD

**Issue by bioinfornatics** [Unvanquished/Unvanquished#730]
Dear,

On linux the old standard location to put local config files was `$HOME/.myApp` while now is `$HOME/.local/share/myApp`

It will be cleaner to do it.

**Comment by Viech** [comment]
That's not precise either. To be XDG-compliant we would need to put *config* files (like keybindings) in `$XDG_CONFIG_HOME`, which defaults to `$HOME/.config` and *data* files (like downloaded maps) in `$XDG_DATA_HOME`, which defaults to `$HOME/.local/share` (see the [XDG Base Directory Specification](http://standards.freedesktop.org/basedir-spec/basedir-spec-latest.html)).

However, while Unvanquished uses a single "home folder" for both data and config files neither folder is the right place, so we could as well stick with just `$HOME`. Note that the updater that's being worked on will download the base game to `$XDG_DATA_HOME`. (@DolceTriade may want to read this post, too.)

In general I'm slightly in favor of splitting data and configuration like that but I wouldn't actually change it unless another change of directory structure becomes necessary because this would annoy users as they had to manually move over their config.

**Comment by DolceTriade** [comment]
Also note that for the updater, it just puts the main install files in ```$XDG_DATA_HOME```. The data is still stored in the old data home. If someone is interested in taking up this task, I wouldn't be against it, but I really would not consider this a priority.

**Comment by illwieckz** [comment]
As an example, I believe that [dhewm3](https://github.com/dhewm/dhewm3) is using `.local/share/dhewm3` as a compromise.

**Comment by Veyrdite** [comment]
What advantage does being XDG compliant here provide?  I understand that XDG provides a lot of standards that are useful, but I've never understood what the pros of this particular one are other than perhaps being a little tidier.

**Comment by DolceTriade** [comment]
You basically summed up the pros and cons Veyrdite. It does not really provide any direct benefit, hence why I don't think this will be a priority from within the team. HOwever, if someone who feels strongly about it were to make a pull request, we wouldn't be against it.

**Comment by Veyrdite** [comment]
I'm against it from a user's point of view.  Keeping everything in one .unvanquished makes things easier to find, backup or transfer.  Every time I've encountered an application splitting its temporary files across multiple directories (sometimes three: .cache , .config and .local/share/) it's given me a headache.  

Of course I'm heavily biased because I'm familiar with/prefer the "old way".

**Comment by TimePath** [comment]
I'm for it from a different view, too many dotfiles in $HOME. All I want is .config/. Split or not split, I don't mind.

**Comment by Viech** [comment]
@Veyrdite, you could modify the `XDG_*` variables to merge directories. That's what they're for.

**Comment by illwieckz** [comment]
I just write something to remind the existence of this issue, which is a good one I think even if not being high priority. Why not just moving `~/.unvanquished` to `~/.local/share/unvanquished` as a first step?

It also raises some questions about the path for non-user data when non-user data are user data… I speak about the ability to get the updater to put things in `~/.local/share/<something>` when the games autodownloads stuff in `~/.local/share/<something>` too.

As a prior implementation (I'm not saying it's a good implementation), QuakeLive faced that issues on Windows putting official assets in `%APPDATA%/id Software/quakelive/baseq3` and user assets in `%APPDATA%/id Software/quakelive/home/baseq3`.

**Comment by illwieckz** [comment]
for another reference, quetoo updater unwraps things like that:

```
<gamepath>/bin/quetoo
<gamepath>/bin/quetoo-update
<gamepath>/lib/libObjectively.so.0
<gamepath>/share/default/map-aerowalk.pk3
<homepath>/.quetoo/default/
```

But I don't know if it's helpful.

**Comment by illwieckz** [comment]
copypasted from irc:

> `<illwieckz>` I don't know if it's a problem to put core game assets (i.e. what updater downloads) in something like .local/share/unvupdater
> `<illwieckz>` and if someone installs unvanquished with his distro package manager in /usr thingy, or using lutris or (futurely) steam
> `<illwieckz>` perhaps it's better to not use the same directory for user content
> `<illwieckz>` if people install unvanquished using playdeb (for example), he would be able to safely rm -rf ~/.local/share/unvudpater

**Comment by illwieckz** [comment]
I about the XDG stuff, well, I don't like how it's done (`.local/share`, `.config`, `.cache` etc.) they just did it wrong, doing it the -"I don't like having many dot files in my home dir, so let's create plenty dot files in my home dir"- but meh, I prefer having things following wrong design the same way and being consistent with this wrong design than having everything doing anything their own own better way.

What do you think about that file layout?

Updater stuff:
```
~/.config/unvanquished/updater.conf
~/.local/share/unvanquished/updater/daemon
~/.local/share/unvanquished/updater/pkg/unvanquished_0.50.0.pk3
```

Game stuff:
```
~/.local/share/unvanquished/user/config/autogen.cfg
~/.local/share/unvanquished/user/config/map/default.cfg
~/.local/share/unvanquished/user/game/maprotation.cfg
~/.local/share/unvanquished/user/pkg/map-hangar28_b1.pk3
~/.local/share/unvanquished/user/screenshots/unvanquished_2017-03-25_090736_000.jpg
```

In fact, just moving the whole `~/.unvanquished/` to `~/.local/share/unvanquished/user/`.

If someone install Unvanquished using some distro package (like in `/usr/share/unvanquished/`), there will be no `~/.local/share/unvanquished/updater/` or if some relics are there, system-wide installation will not use it.

I don't thing moving `~/.unvanquished/config/` to something like `~/.config/unvanquished/` because the way "config" is done in unvanquished is not very "config", like map rotation list etc. they are kind of scripts, and well even config files are loaded by an "exec" command. These files are half part config, half part code, half part asset (yes it means 3 half parts, that's a zombie). 

If one day we have some stuff to cache out (like runtime generated navmesh and minimap), we would be able to put them in `~/.cache/unvanquished/navmesh/` and `~/.cache/unvanquished/minimap/` for example.

If you have a better idea for `user` in `~/.local/share/unvanquished/user/` feel free to make a suggestion.

**Comment by illwieckz** [comment]
Note: it's my own taste, but I don't like paths like `~/.config/Unvanquished Development/Unvanquished Updater.conf` with mixed case and spaces. I remember years ago we moved from `~/.Unvanquished` to `~/.unvanquished` and that pleased me.

**Comment by illwieckz** [comment]
This is the current implementation:

Updater stuff: Unvanquished/updater2#13 

```
~/.config/unvanquished/updater.conf
~/.local/share/unvupdater/daemon
~/.local/share/unvupdater/pkg/unvanquished_0.50.0.pk3
```
Game stuff: DaemonEngine/Daemon#20 

```
~/.local/share/unvanquished/config/autogen.cfg
~/.local/share/unvanquished/config/map/default.cfg
~/.local/share/unvanquished/game/maprotation.cfg
~/.local/share/unvanquished/pkg/map-hangar28_b1.pk3
~/.local/share/unvanquished/screenshots/unvanquished_2017-03-25_090736_000.jpg
```

**Comment by Viech** [comment]
I think I like your layout (the "current implementation in unmerged PR").

**Comment by illwieckz** [comment]
Fixed in engine in `for-0.51.0` branch.
